lib/assets_cache.dart

```diff
@@ -1,31 +1,64 @@
 import 'package:flutter/services.dart' show rootBundle;
+import 'dart:typed_data';
 
 /// A class that loads, and cache files
 ///
 /// it automatically looks for files on the assets folder
 class AssetsCache {
-  Map<String, String> textFiles = {};
+  final Map<String, _Asset> _files = {};
 
   /// Removes the file from the cache
   void clear(String file) {
-    textFiles.remove(file);
+    _files.remove(file);
   }
 
   /// Removes all the files from the cache
   void clearCache() {
-    textFiles.clear();
+    _files.clear();
   }
 
   /// Reads a file from assets folder
   Future<String> readFile(String fileName) async {
-    if (!textFiles.containsKey(fileName)) {
-      textFiles[fileName] = await _readFile(fileName);
+    if (!_files.containsKey(fileName)) {
+      _files[fileName] = await _readFile(fileName);
     }
 
-    return textFiles[fileName];
+    assert(
+        _files[fileName] is _StringAsset, '"$fileName" is not a String Asset');
+
+    return _files[fileName].value;
   }
 
-  Future<String> _readFile(String fileName) async {
-    return await rootBundle.loadString('assets/$fileName');
+  /// Reads a binary file from assets folder
+  Future<List<int>> readBinaryFile(String fileName) async {
+    if (!_files.containsKey(fileName)) {
+      _files[fileName] = await _readBinary(fileName);
+    }
+
+    assert(
+        _files[fileName] is _BinaryAsset, '"$fileName" is not a Binary Asset');
```

Comment:
for both asserts. either all in one line or

```dart
assert(
    foo,
    bar,
);
```

Reply:
THL